File: javascript/matrix/matrix.js
//
// This is only a SKELETON file for the 'Matrix' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Matrix {
  
  constructor(rowString) {
    let rawRows = rowString.split('\n');
    let tmpRows = [];
    rawRows.forEach(element => {
      let tmpCol = [];
      let tmpCols = element.split(' ');
      tmpCols.forEach(c => tmpCol.push(Number(c)));
      tmpRows.push(tmpCol);
    });
    this.matrix = tmpRows;
  }

  get rows() {
    return this.matrix;
  }

  get columns() {
    let shiftedMatrix = [];
    for (let i=0; i < this.matrix[0].length; i++) {
      let tmpRow = [];
      for (let j=0; j < this.matrix.length; j++) {
        tmpRow.push(this.matrix[j][i]);
      }
      shiftedMatrix.push(tmpRow);
    }
    return shiftedMatrix;
  }
}

File: ruby/matrix/matrix.rb

class Matrix

    def initialize(matString)
        @matrix = Array(matString.split("\n").map { |s|
            Array(s.split(" ").map(&:to_i))
        })
    end

    def rows
        return @matrix
    end

    def columns
        @columns = []
        for i in 0..@matrix[0].length-1
            @columns = @columns.push([])
            for j in 0..@matrix.length-1
                @columns[i] = @columns[i].push(@matrix[j][i])
            end
        end
        return @columns
    end
end
